apps/credential-showcase-api-server/src/database/repositories/__tests__/persona.repository.test.ts

```diff
@@ -0,0 +1,148 @@
+      headshotImageId: '456',
+    }
+
+    const savedPersona = await repository.create(persona)
+
+    expect(savedPersona).toBeDefined()
+    expect(savedPersona.name).toStrictEqual(persona.name)
+    expect(savedPersona.role).toStrictEqual(persona.role)
+    expect(savedPersona.description).toStrictEqual(persona.description)
+    expect(savedPersona.bodyImageId).toStrictEqual(persona.bodyImageId)
+    expect(savedPersona.headshotImageId).toStrictEqual(persona.headshotImageId)
+  })
+
+  it('Should get persona by id from database', async (): Promise<void> => {
+    const persona: NewPersona = {
+      name: 'test',
+      role: 'tester',
+      description: 'some persona',
+    }
+
+    const savedPersona = await repository.create(persona)
+    expect(savedPersona).toBeDefined()
+
+    const fromDb = await repository.findById(savedPersona.id)
+
+    expect(fromDb).not.toBeNull()
+    expect(fromDb!.name).toStrictEqual(persona.name)
+    expect(fromDb!.role).toStrictEqual(persona.role)
+    expect(fromDb!.description).toStrictEqual(persona.description)
+  })
+
+  it('Should get all personas from database', async (): Promise<void> => {
+    const persona1: NewPersona = {
+      name: 'test',
+      role: 'tester',
+      description: 'some persona',
+    }
+
+    const savedPersona1 = await repository.create(persona1)
+    expect(savedPersona1).toBeDefined()
+
+    const persona2: NewPersona = {
+      name: 'test',
+      role: 'tester',
+      description: 'some persona',
+    }
+
+    const savedAsset2 = await repository.create(persona2)
+    expect(savedAsset2).toBeDefined()
+
+    const fromDb = await repository.findAll()
+
+    expect(fromDb.length).toStrictEqual(2)
+  })
+
+  it('Should delete persona from database', async (): Promise<void> => {
+    const persona: NewPersona = {
+      name: 'test',
+      role: 'tester',
+      description: 'some persona',
+    }
+
+    const savedPersona = await repository.create(persona)
+    expect(savedPersona).toBeDefined()
+
+    const deletionResult = await repository.delete(savedPersona.id)
+    expect(deletionResult).toStrictEqual(true)
+
+    const fromDb = await repository.findById(savedPersona.id)
+    expect(fromDb).toBeNull()
+  })
+
+  it('Should update persona in database', async (): Promise<void> => {
+    const persona: NewPersona = {
+      name: 'test',
+      role: 'tester',
+      description: 'some persona',
+    }
+
+    const savedPersona = await repository.create(persona)
+    expect(savedPersona).toBeDefined()
+
+    const newRole = 'senior tester'
+    const updatedPersona = await repository.update(savedPersona.id, {
+      name: savedPersona.name,
+      role: newRole,
+      description: savedPersona.description || undefined,
+    })
+
+    expect(updatedPersona).toBeDefined()
+    expect(updatedPersona.name).toStrictEqual(persona.name)
+    expect(updatedPersona.role).toStrictEqual(newRole)
+    expect(updatedPersona.description).toStrictEqual(persona.description)
+  })
```

Comment:
maybe add a test what happens when you create and update a persona with an image id that does not exist